### ISSUE TITLE
OHLUX GU24 Smart Bulb Connection to device succeeded but no datapoints found

### ISSUE THREAD

**Issue by seeshaughnessy** [rospogrigio/localtuya#1748]
## The problem
I'm trying to add two OHLUX GU24 smart bulbs (https://a.co/d/07JN8T2z), but they don't show up on the configuration list, and manually adding them shows the error:

"Connection to device succeeded but no datapoints found, please try again. Create a new issue and include debug logs if problem persists."

## Environment

- Localtuya version: 5.2.1
- Home Assistant Core version: 2024.7.2
- [] Does the device work using the Home Assistant Tuya Cloud component ?
- [] Does the device work using the Tinytuya (https://github.com/jasonacox/tinytuya) command line tool ? 
- [] Was the device working with earlier versions of localtuya ? Which one? 
- [X] Are you using the Tuya/SmartLife App in parallel ?

## Steps to reproduce

1. In local tuya, press "CONFIGURE"
2. "Add New Device"
3. Select the "..."
4. Filled out Device details and submitted.

## Provide Home Assistant traceback/logs

```
Logger: homeassistant
Source: custom_components/localtuya/discovery.py:67
integration: LocalTuya (documentation, issues)
First occurred: 1:55:56 PM (104 occurrences)
Last logged: 2:00:11 PM

Error doing job: Exception in callback _SelectorDatagramTransport._read_ready() (None)
Traceback (most recent call last):
  File "/config/custom_components/localtuya/discovery.py", line 65, in datagram_received
    data = decrypt_udp(data)
           ^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/discovery.py", line 30, in decrypt_udp
    return _unpad(decryptor.update(message) + decryptor.finalize()).decode()
                                              ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/cryptography/hazmat/primitives/ciphers/base.py", line 184, in finalize
    data = self._ctx.finalize()
           ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/cryptography/hazmat/backends/openssl/ciphers.py", line 223, in finalize
    raise ValueError(
ValueError: The length of the provided data is not a multiple of the block length.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "/usr/local/lib/python3.12/asyncio/selector_events.py", line 1238, in _read_ready
    self._protocol.datagram_received(data, addr)
  File "/config/custom_components/localtuya/discovery.py", line 67, in datagram_received
    data = data.decode()
           ^^^^^^^^^^^^^
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xc2 in position 0: invalid continuation byte
```

and

```
Logger: custom_components.localtuya.pytuya
Source: custom_components/localtuya/pytuya/__init__.py:863
integration: LocalTuya (documentation, issues)
First occurred: 2:00:01 PM (1 occurrences)
Last logged: 2:00:01 PM

[eb2...uqp] Failed to get status:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/locks.py", line 386, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
    async with timeouts.timeout(timeout):
  File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
```

**Comment by 2rym** [comment]
I'm struggling with this exact problem

**Comment by thannerfabian** [comment]
I have the same problem, but for switches only. For humidity and temperature sensors, it worked.

**Comment by martinw72** [comment]
the same here, started with HA 2024.7.2!

**Comment by thannerfabian** [comment]
@martinw72 can you confirm that it worked with the version before 2024.7.2?

**Comment by martinw72** [comment]
@thannerfabian I'm not sure, because I've not change my configuration, but get 10000 times the exact same entry I've not seen before 2024.7.2. But maybe my update and the flooded error logs are Independent of each other. My experience tells me that there is a connection between this two!

**Comment by thannerfabian** [comment]
@martinw72 did you already update to 2024.7.3?

**Comment by martinw72** [comment]
Yes, I’m on 2024.7.3!

**Comment by regevbr** [comment]
Exact same issue for me

```
[bf6...nxy] Failed to get status:
Traceback (most recent call last):
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 520, in wait_for
    return await fut
           ^^^^^^^^^
  File "/usr/local/lib/python3.12/asyncio/locks.py", line 386, in acquire
    await fut
asyncio.exceptions.CancelledError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 863, in detect_available_dps
    data = await self.status()
           ^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 795, in status
    status = await self.exchange(DP_QUERY)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 769, in exchange
    msg = await self.dispatcher.wait_for(seqno, payload.cmd)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/localtuya/pytuya/__init__.py", line 450, in wait_for
    await asyncio.wait_for(self.listeners[seqno].acquire(), timeout=timeout)
  File "/usr/local/lib/python3.12/asyncio/tasks.py", line 519, in wait_for
    async with timeouts.timeout(timeout):
  File "/usr/local/lib/python3.12/asyncio/timeouts.py", line 115, in __aexit__
    raise TimeoutError from exc_val
TimeoutError
```

**Comment by balrog-kun** [comment]
I've hit this issue too, done some debugging.

TL;DR: try uninstalling this extension and using the following, improved version: https://github.com/xZetsubou/hass-localtuya

Long version: The error is caused by Tuya 3.5 devices. localtuya supports protocols up to 3.4 right now. UDP message format and encryption is different. There may be more differences.

I added the decryption of the 3.5 discovery packets to my repo but when looking for where to contribute the changes, I found this respository (rospogrigio/localtuya) has been abandonded for about a year with lots of improvements pending review from the maintainer. It looks like xZetsubou/hass-localtuya is currently the most complete fork and it looks like it has protocol 3.5 support already.
